### DISCUSSION TITLE
Convergence of primal solver

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#452] [General]
Hi

I have  observed a behaviour and want to understand why it is happening.

Case: periodic hill field inversion
Optimizer: IPOPT
Solver: DASimpleFoam
DAFoam version: v3.0.3

I set maximum iterations for the primal sovler to be 20000 and i set the primalResidualTolerace to be 1e-06.

I encounter residual stalling at some point during the optimization process (keep in mind, that my pure CFD case convegres well and I use the same settings for DAFoam to do the field inversion). 

When this residual stalling is of the order of 1e-05 or O(10^-6), then the primal solver maxes out at 20000 and either does a line search or moves on to the next optimization iteration (if there is sufficient decrease of the barrier objective function).

But, when the residual is of the order 10^-4 or lower, i get a message saying primal solution failed! 

Now technically, primal solution failed should be reported regardless of whether the residual is 1e-05 or 1e-04 or 1e-03 simply because it is all higher than the residual tolerance of 1e-06 that i set, correct? 

What could be the explanation for such behavior and are the results reliable? I need to mention that this problem happens with pressure only.

**Comment by friedenhe** [comment]
Check https://dafoam.github.io/mydoc_get_started_faq.html#how-to-fix-the-primal-solution-failed-for-the-baseline-design-error
